content/en/blog/_posts/2023-09-12-comparing-local-kubernetes-development-tools.md

```diff
@@ -0,0 +1,50 @@
+---
+layout: blog
+title: 'Comparing Local Kubernetes Development Tools: Telepresence, Gefyra, and mirrord'
+date: 2023-09-12
+slug: local-k8s-development-tools
+---
+
+**Author:** Eyal Bukchin (MetalBear)
+
+The Kubernetes development cycle is an evolving landscape with a myriad of tools seeking to streamline the process. Each tool has its unique approach, and the choice often comes down to individual project requirements, the team's expertise, and the preferred workflow.
+
+Among the various solutions, a category we dubbed “Local K8S Development tools” has emerged, which seeks to enhance the Kubernetes development experience by connecting locally running components to the Kubernetes cluster. This facilitates rapid testing of new code in cloud conditions, circumventing the traditional cycle of Dockerization, CI, and deployment.
+
+In this post, we compare three solutions in this category: Telepresence, Gefyra, and our own contender, mirrord.
+
+## Telepresence
+The oldest and most well-established solution in the category, [Telepresence](https://www.telepresence.io/) uses a VPN (or more specifically, a TUN device) to connect the user's machine (or a locally running container) and the cluster's network. It then supports the interception of incoming traffic to a specific service in the cluster, and its redirection to a local port. The traffic being redirected can also be filtered to avoid completely disrupting the remote service. It also offers complementary features to support file access (by locally mounting a volume mounted to a pod) and importing environment variables.
+Telepresence requires the installation of a local daemon (which requires root privileges) and a Traffic Manager component on the cluster. Additionally, it runs an Agent as a sidecar on the pod to intercept the desired traffic.
+
+
+
+## Gefyra
+[Gefyra](https://gefyra.dev/), similar to Telepresence, employs a VPN to connect to the cluster but operates at the Docker network level, thus only supporting containers. This approach enhances portability across different OSes and local setups. However, the downside is that it does not support natively run uncontainerized code.
+
+Gefyra primarily focuses on network traffic, leaving file access and environment variables unsupported. Unlike Telepresence, it doesn't alter the workloads in the cluster, ensuring a straightforward clean-up process if things go awry.
```

Comment:
_Docker_ network? Modern clusters usually don't have a Docker network. Can you explain more about where Docker is involved and how that's relevant to people who use Kubernetes?